File: android/Ninehoto.Android/Utilities/MockDataGenerator.cs
using System;
using System.Collections.Generic;
using Ninehoto.Android.Models;

namespace Ninehoto.Android.Utilities
{
    public class MockDataGenerator
    {
        private static MockDataGenerator? _instance;
        public static MockDataGenerator Instance => _instance ??= new MockDataGenerator();

        private MockDataGenerator() { }

        public List<MediaItem> GenerateMockMediaItems(int count, bool includeVideos = true)
        {
            var items = new List<MediaItem>();

            for (int i = 0; i < count; i++)
            {
                var isVideo = includeVideos && (i % 5 == 0);
                var item = new MediaItem
                {
                    Id = $"mock-asset-{i}-{Guid.NewGuid()}",
                    MediaType = isVideo ? MediaItem.MediaType.Video : MediaItem.MediaType.Photo,
                    CreationDate = DateTime.Now.AddDays(-i),
                    PixelWidth = new Random().Next(1000, 4000),
                    PixelHeight = new Random().Next(1000, 4000),
                    IsVideo = isVideo
                };

                if (isVideo)
                {
                    item.Duration = new Random().Next(5, 300);
                }

                items.Add(item);
            }

            return items;
        }

        public List<SessionStatistics> GenerateMockStatistics(int count)
        {
            var sessions = new List<SessionStatistics>();
            var random = new Random();

            for (int i = 0; i < count; i++)
            {
                var swipeCount = random.Next(20, 200);
                var deleteCount = random.Next(0, swipeCount);
                var keepCount = swipeCount - deleteCount;
                var duration = random.Next(60, 600);

                var session = new SessionStatistics
                {
                    SwipeCount = swipeCount,
                    DeleteCount = deleteCount,
                    KeepCount = keepCount,
                    SessionDate = DateTime.Now.AddDays(-i * 7),
                    DurationSeconds = duration,
                    AverageSwipesPerMinute = swipeCount / (duration / 60.0)
                };

                sessions.Add(session);
            }

            return sessions;
        }

        public UserPreferences GenerateMockUserPreferences()
        {
            var random = new Random();
            var qualityValues = Enum.GetValues(typeof(ThumbnailQuality));
            var sortValues = Enum.GetValues(typeof(SortOrder));

            return new UserPreferences
            {
                HapticFeedbackEnabled = random.Next(2) == 1,
                ThumbnailQuality = (ThumbnailQuality)qualityValues.GetValue(random.Next(qualityValues.Length)),
                SortOrder = (SortOrder)sortValues.GetValue(random.Next(sortValues.Length)),
                HasCompletedOnboarding = true,
                LastSessionDate = DateTime.Now.AddDays(-random.Next(0, 7))
            };
        }
    }

    public class UserPreferences
    {
        public bool HapticFeedbackEnabled { get; set; }
        public ThumbnailQuality ThumbnailQuality { get; set; }
        public SortOrder SortOrder { get; set; }
        public bool HasCompletedOnboarding { get; set; }
        public DateTime? LastSessionDate { get; set; }
    }
}

File: ios/Ninehoto/Utilities/MockDataGenerator.swift
import Foundation
import Photos

final class MockDataGenerator {
    static let shared = MockDataGenerator()

    private init() {}

    func generateMockAssets(count: Int, includeVideos: Bool = true) -> [PHAsset] {
        var assets: [PHAsset] = []

        for i in 0..<count {
            let asset = PHAsset()
            let identifier = "mock-asset-\(i)-\(UUID().uuidString)"

            asset.setValue(identifier, forKey: "localIdentifier")

            let isVideo = includeVideos && (i % 5 == 0)
            asset.setValue(isVideo ? PHAssetMediaType.video : PHAssetMediaType.image, forKey: "mediaType")

            let daysAgo = Double(i)
            let date = Date().addingTimeInterval(-daysAgo * 86400)
            asset.setValue(date, forKey: "creationDate")

            let width = Int.random(in: 1000...4000)
            let height = Int.random(in: 1000...4000)
            asset.setValue(width, forKey: "pixelWidth")
            asset.setValue(height, forKey: "pixelHeight")

            if isVideo {
                let duration = Double.random(in: 5...300)
                asset.setValue(duration, forKey: "duration")
            }

            assets.append(asset)
        }

        return assets
    }

    func generateMockMediaItems(count: Int, includeVideos: Bool = true) -> [MediaItem] {
        let assets = generateMockAssets(count: count, includeVideos: includeVideos)
        return assets.map { MediaItem(asset: $0) }
    }

    func generateMockStatistics(count: Int) -> [SessionStatistics] {
        var sessions: [SessionStatistics] = []

        for i in 0..<count {
            let swipeCount = Int.random(in: 20...200)
            let deleteCount = Int.random(in: 0...swipeCount)
            let keepCount = swipeCount - deleteCount
            let duration = Double.random(in: 60...600)

            let session = SessionStatistics(
                swipeCount: swipeCount,
                deleteCount: deleteCount,
                keepCount: keepCount,
                sessionDate: Date().addingTimeInterval(-Double(i) * 86400 * 7),
                durationSeconds: duration,
                averageSwipesPerMinute: swipeCount / (duration / 60)
            )
            sessions.append(session)
        }

        return sessions
    }

    func generateMockUserPreferences() -> UserPreferences {
        UserPreferences(
            hapticFeedbackEnabled: Bool.random(),
            thumbnailQuality: PreferencesManager.ThumbnailQuality.allCases.randomElement() ?? .medium,
            sortOrder: PreferencesManager.SortOrder.allCases.randomElement() ?? .dateDescending,
            hasCompletedOnboarding: true,
            lastSessionDate: Date().addingTimeInterval(-Double.random(in: 0...604800))
        )
    }
}

struct UserPreferences: Codable {
    let hapticFeedbackEnabled: Bool
    let thumbnailQuality: PreferencesManager.ThumbnailQuality
    let sortOrder: PreferencesManager.SortOrder
    let hasCompletedOnboarding: Bool
    let lastSessionDate: Date?
}